# Lab 10

## Exercise 1:

Let's have a go with the BBC website. Load the BBC News page and print out all of the links, then write a loop that visits and prints out the news titles in the page of each link.

In [3]:

from bs4 import BeautifulSoup

import requests
response = requests.get("https://bbc.com/news")
data = response.text

data_bs = BeautifulSoup(data)

list_of_links = data_bs.find_all('a')

print('The list of links:',list_of_links)

for l in list_of_links:
    link=l.get('href')
    print("Link:",link,l.text)
# for l in list_of_links:
#   link=l.get('href')
#   # response = requests.get("http://www.mathpractice.xyz"+link)
#   # new_page = response.text
#   print("New page:","http://www.mathpractice.xyz"+link,l.text)


The list of links: [<a class="sc-2e6baa30-0 gILusN" data-testid="internal-link" href="/"><!--$--><svg category="logo" class="sc-1097f7fe-0 jbvZzi" icon="bbc" viewbox="0 0 112 32" xmlns="http://www.w3.org/2000/svg"><title>British Broadcasting Corporation</title><path d="M111.99999,4.44444577e-05 L111.99999,32.0000444 L79.9999905,32.0000444 L79.9999905,4.44444577e-05 L111.99999,4.44444577e-05 Z M72.0000119,-3.55271368e-15 L72.0000119,32 L40.0000119,32 L40.0000119,-3.55271368e-15 L72.0000119,-3.55271368e-15 Z M32,-3.55271368e-15 L32,32 L-1.13686838e-13,32 L-1.13686838e-13,-3.55271368e-15 L32,-3.55271368e-15 Z M97.469329,6.80826869 C96.0294397,6.80826869 94.7294393,7.02226876 93.5693278,7.44982444 C92.4089942,7.87782457 91.4137717,8.49471364 90.5841047,9.30049166 C89.7538823,10.1067141 89.1188821,11.07327 88.6785486,12.199937 C88.2378818,13.3269373 88.0177706,14.5896043 88.0177706,15.9876048 C88.0177706,17.4188274 88.2296596,18.7062722 88.6531042,19.8493837 C89.0763265,20.9929396 89.686104

## Exercise 2: (Quiz)

Extending the above exercise: write a loop inside the above loop that visits the second level pages. That is:
- Outer loop: visits pages 1, 2, 3, .. from the main page
- Inner loop: visits pages 1.1, 1.2, 1.3, .. from page 1, then 2.1, 2.2, 2.3, .. from page 2, etc.

## Exercise 3:

Now, pick a news article, then view its source. Can you find the news text in the source?





## Exercise 4:

Read the page `https://www.mathpractice.xyz/year12/skills` using `bs4` and write a loop that prints out all of the link texts under "Algebra".

## Exercise 5:

Read the page `https://www.mathpractice.xyz` using `bs4` and write a loop that prints out all of the image tags in the page.

## Exercise 6:

Read the page `https://www.mathpractice.xyz/algebra` using `bs4` and then write a loop that opens each link in the page and saves it into a file. Name the file after the link title.